storage/innobase/srv/srv0srv.cc

```diff
@@ -3685,139 +3680,174 @@ srv_enable_redo_encryption_rk(THD* thd)
-				<< ", since innodb_undo_tablespaces=0.";
-			return;
-		}
-		if (srv_read_only_mode) {
-			srv_undo_log_encrypt = false;
-			ib::error() << cant_set_undo_tablespace << "s"
-				<< to_be_encrypted
-				<< " in read-only mode.";
-			return;
+	if (srv_undo_tablespaces == 0) {
+		srv_undo_log_encrypt = false;
+		ib::error() << "Can't set undo tablespace" << "s"
+			<< " to be encrypted"
+			<< ", since innodb_undo_tablespaces=0.";
+
+                ib_senderrf(
+                                thd, IB_LOG_LEVEL_WARN,
+                                ER_UNDO_ENCRYPTION_ERROR,
+		                "Can't set undo tablespaces"
+			        " to be encrypted"
+			        ", since innodb_undo_tablespaces=0."
+                );
+
+		return true;
+	}
+	if (srv_read_only_mode) {
+		srv_undo_log_encrypt = false;
+		ib::error() << "Can't set undo tablespace" << "s"
+			<<" to be encrypted" 
+			<< " in read-only mode.";
+                ib_senderrf(
+                                thd, IB_LOG_LEVEL_WARN,
+                                ER_UNDO_ENCRYPTION_ERROR,
+		                "Can't set undo tablespaces"
+			        " to be encrypted"
+			        ", since innodb_undo_tablespaces=0."
+                );
+		return true;
+	}
+	ulint undo_spaces[TRX_SYS_N_RSEGS + 1];
+	const ulint undo_spaces_no = trx_rseg_get_n_undo_tablespaces(undo_spaces);
+	for (ulint undo_idx = 0; undo_idx < undo_spaces_no; ++undo_idx)
+	{
+		/* Skip system tablespace, since it's also shared
+		   tablespace. */
+		const ulint space_id = undo_spaces[undo_idx];
+		if (space_id == TRX_SYS_SPACE) {
+			continue;
 		}
-		ulint undo_spaces[TRX_SYS_N_RSEGS + 1];
-		const ulint undo_spaces_no = trx_rseg_get_n_undo_tablespaces(undo_spaces);
-		for (ulint undo_idx = 0; undo_idx < undo_spaces_no; ++undo_idx)
-		{
-			/* Skip system tablespace, since it's also shared
-			   tablespace. */
-			const ulint space_id = undo_spaces[undo_idx];
-			if (space_id == TRX_SYS_SPACE) {
-				continue;
-			}
-			space = fil_space_get(space_id);
-			ut_ad(fsp_is_undo_tablespace(space_id));
-			/* This flag will be written to the header
-			   later, by calling the fsp_header_write_encryption()
+		space = fil_space_get(space_id);
+		ut_ad(fsp_is_undo_tablespace(space_id));
+		/* This flag will be written to the header
+		   later, by calling the fsp_header_write_encryption()
 function: */
-			ulint	new_flags =
-				space->flags | FSP_FLAGS_MASK_ENCRYPTION;
-			/* We need the server_uuid initialized, otherwise,
-			   the keyname will not contains server uuid. */
-			if (FSP_FLAGS_GET_ENCRYPTION(space->flags)
-					|| strlen(server_uuid) == 0) {
-				continue;
+		ulint	new_flags =
+			space->flags | FSP_FLAGS_MASK_ENCRYPTION;
+		/* We need the server_uuid initialized, otherwise,
+		   the keyname will not contains server uuid. */
+		if (FSP_FLAGS_GET_ENCRYPTION(space->flags)
+				|| strlen(server_uuid) == 0) {
+			continue;
+		}
+		dberr_t err;
+		mtr_t	mtr;
+		byte	encrypt_info[ENCRYPTION_INFO_SIZE_V2];
```

Comment:
interestingly, I could not find ENCRYPTION_INFO_SIZE_V2 in the source code, just ENCRYPTION_INFO_SIZE.